### PR TITLE
[Segment] Scroll depth tracking

### PR DESCRIPTION
**Pull request by TerryDevS**
### Description
[Asana](https://app.asana.com/0/1200268323978783/1203049426090926/f)

### Staging Docs

http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/segment-scrolldepth-tracking

### CODE REVIEW

File: assets/js/scrollDepth.js [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+  const initObserver = () => {
+    const cb = (entries) => {
+    const [IntersectionObserverEntry] = entries;
+    const { isIntersecting, target } = IntersectionObserverEntry;
+    const position = Number(target.dataset.depthMarkerId);
 
       if (isIntersecting) {
-       analytics.track('Scroll Depth', {
+        analytics.track('Scroll Depth', {
           scrollDepth: position === 99 ? 100 : position // triggering at 99% is more consistent than 100%.
```
Comment (davetroiano) [inline assets/js/scrollDepth.js:17]: can position be floating pt? and should this instead be `position > 99 ? 100 : position`

**Review by davetroiano — APPROVED** [review]
seems good to me though @hartantothio or @scott-confluent could you please have a look?

File: assets/js/scrollDepth.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+const SCROLL_DEPTH_LEVELS = [20, 40, 60, 80, 99];
+
+window.addEventListener('DOMContentLoaded', function () {
+  let observer
+  const el = document.getElementsByTagName('body')[0];
```
Comment (rafikalid) [inline assets/js/scrollDepth.js:5]: why not using just `document.body` ?

File: assets/js/scrollDepth.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    
+  el.style.position = 'relative';
```
Comment (rafikalid) [inline assets/js/scrollDepth.js:7]: Maybe doing this in css instead?

File: assets/js/scrollDepth.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+  const initObserver = () => {
+    const cb = (entries) => {
+    const [IntersectionObserverEntry] = entries;
```
Comment (rafikalid) [inline assets/js/scrollDepth.js:11]: I don't see `entries` anywhere else, I prefer doing destruction directly in arguments
`( [IntersectionObserverEntry] ) =>` instead

File: assets/js/scrollDepth.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    const { isIntersecting, target } = IntersectionObserverEntry;
```
Comment (rafikalid) [inline assets/js/scrollDepth.js:12]: Do it in args I prefer, it simpler I think
`( [ { isIntersecting, target } ] ) =>`

File: assets/js/scrollDepth.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    const position = Number(target.dataset.depthMarkerId);
```
Comment (rafikalid) [inline assets/js/scrollDepth.js:13]: we need this inside the `if` block only, could you please put it there

File: assets/js/scrollDepth.js (hunk 1 above)
Comment (rafikalid) [reply assets/js/scrollDepth.js:17]: @davetroiano the position here is a hard coded data, not the scroll position it self.

File: assets/js/scrollDepth.js (hunk 5 above)
Comment (TerryDevS) [reply assets/js/scrollDepth.js:12]: It is simpler but not better for readability IMO. but I can destructure it.

File: assets/js/scrollDepth.js (hunk 6 above)
Comment (TerryDevS) [reply assets/js/scrollDepth.js:13]: sorry I am not following

File: assets/js/scrollDepth.js (hunk 4 above)
Comment (TerryDevS) [reply assets/js/scrollDepth.js:11]: sorry I am not following

**Review by rafikalid — APPROVED** [review]

**Pull request merged**